ORM/DataList.php

```diff
@@ -403,12 +405,10 @@ public function filter() {
 	public function addFilter($filterArray) {
 		$list = $this;
 
-		foreach($filterArray as $field => $value) {
-			$fieldArgs = explode(':', $field);
-			$field = array_shift($fieldArgs);
-			$filterType = array_shift($fieldArgs);
-			$modifiers = $fieldArgs;
-			$list = $list->applyFilterContext($field, $filterType, $modifiers, $value);
+		foreach($filterArray as $expression => $value) {
+			list($field, $filterClass, $modifiers) = $this->parseFilterArguments($expression);
+			$filter = Injector::inst()->create($filterClass, $field, $value, $modifiers);
```

Comment:
It makes more sense to me to have a single field that take `$key, $value` as an argument and returns a configured SearchFilter object.

E.g.: `$filter = $this->createSearchFilter($expression, value)`

This should work in the 2nd call.